content/ADR-284-private-voice-chat.md

```diff
@@ -0,0 +1,362 @@
+    Ok ok = 1;
+    InternalServerError internal_server_error = 2;
+    NotFoundError not_found = 4;
+    ForbiddenError forbidden_error = 5;
+  }
+}
+
+rpc AcceptPrivateVoiceChat(AcceptPrivateVoiceChatPayload) returns (AcceptPrivateVoiceChatResponse) {}
+```
+
+### Ending a private voice chat
+
+Ending a private voice chat can be done through the Social Service in any of their states (ringing or on going) or through LiveKit if the user is already on an on going private voice chat.
+
+#### Ending a private voice chat through the Social Service
+
+Ending a private voice chat is done by performing the `EndPrivateVoiceChat` RPC call with the call id of the voice chat.
+A voice chat can be ended at any state by any of the participants, but the callee should use the reject operation if they want to reject a private voice chat.
+This process will result in the caller will be notified through the updates subscription mechanism mentioned in the _Subscribing private voice chats updates_ section with the call id that is being terminated.
+
+This RPC call can be used in two different states of a private voice chat:
+
+1. The voice chat is in the ringing state.
+2. The user is in an on going private voice chat (connected to LiveKit).
+
+For each of these states the client must react to them in the following way:
+
+- In the first case, the ringing process must be ended.
+- In the second case, the voice chat must be ended by issuing a voluntarily disconnection from LiveKit.
+
+Here's the interaction flow of the Explorer and the Social Service:
+
+```mermaid
+sequenceDiagram
+	actor U1 as Caller or calle
+	participant SS as Social Server
+	participant LK as LiveKit
+
+  U1->>SS: End private voice chat with id
+  break Private voice chat exists
+		SS-->>U1: Ok
+	end
+	LK->>U1: Room destroyed
+```
+
+```protobuf
+message EndPrivateVoiceChatPayload {
+  string call_id = 1;
+}
+
+message EndPrivateVoiceChatResponse {
+  message Ok {
+    string call_id = 1;
+  }
+
+  oneof response {
+    Ok ok = 1;
+    InternalServerError internal_server_error = 2;
+    NotFoundError not_found = 3;
+  }
+}
+
+rpc EndPrivateVoiceChat(EndPrivateVoiceChatPayload) returns (EndPrivateVoiceChatResponse) {}
+```
+
+#### Ending a private voice chat through LiveKit
+
+An on going LiveKit private voice chat can be ended by issuing a voluntarily disconnection from the room.
+This action is monitored by our services and will result in the LiveKit room being destroyed.
+
+The client must be listening to the LiveKit room events to detect the disconnection reaction.
+In this case, the reason for the disconnection will be set as the destruction of the room, which the client must treat as the end of the private voice chat.
+
+### Getting incoming private voice chats
+
+At least for now, we can't know if a user is connecter or not to the Social Service, therefore we can't strictly enforce the requirement to prevent users from starting private voice chats with disconnected users.
```

Comment:
```suggestion
At least for now, we can't know if a user is connected or not to the Social Service; therefore, we can't strictly enforce the requirement to prevent users from starting private voice chats with disconnected users.
```

Reply:
Changed it!